MetaMask/metamask-extension
# fix: Improve performance of getSelectedAccount

## **Description**

When exploring the extension performance, I spotted that the `getSelectedAccount` selector that was introduced for Multichain was returning a new object, thus the new object reference causing unnecessary re-renders of components using it.  The top-most UI component, `routes.container.js`, was using this selector, which could cause a significant UI perf impact.  This PR memoizes that result, ensuring `routes.container.js` doesn't make unnecessary re-renders.

[![Open in GitHub Codespaces](https://github.com/codespaces/badge.svg)](https://codespaces.new/MetaMask/metamask-extension/pull/24745?quickstart=1)

## **Related issues**

Fixes: N/A

## **Manual testing steps**

*BEFORE CHECKING OUT THIS PR*,  open `routes.component.js` and add the following in `componentDidUpdate`:

```
console.log('[routes.component.js] componentDidUpdate ---------------------------');
    const keyz = Object.keys(this.props);
    keyz.forEach((key) => {
      if (this.props[key] !== prevProps[key]) {
        console.warn('[prop value inequality]: ', key, this.props[key], prevProps[key]);
      }
    });
```

See that the `account` property displays in this list.  This means an unnecessary update is happening.

After checking out this PR, that key should not display in the warning.

## **Screenshots/Recordings**



### **Before**



### **After**



## **Pre-merge author checklist**


## **Pre-merge reviewer checklist**
